hathor/transaction/storage/rocksdb_storage.py

```diff
@@ -151,8 +151,16 @@ def get_all_transactions(self) -> Iterator['BaseTransaction']:
             yield tx
 
     def get_count_tx_blocks(self) -> int:
-        keys_bcount = self._db.get_property(b'rocksdb.estimate-num-keys', self._cf_tx)
-        keys_count = int(keys_bcount)
+        try:
+            # We know RocksDB estimation for the number of keys is not 100% accurate, so we prefer to
+            # count using the indexes, if available
+            keys_count = self.get_tx_count() + self.get_block_count()
+        except NotImplementedError:
+            # This will happen if self.with_index == False, and the only way out is to fallback
+            # to RocksDB key estimation anyway
+            keys_bcount = self._db.get_property(b'rocksdb.estimate-num-keys', self._cf_tx)
```

Comment:
As this returns an estimation instead of the exact value, I feel we should really use it as a fallback unless authorized by the caller. Or we can just move it to another method and let the caller decide what's best for their usage.

Reply:
Couldn't we just move `get_count_tx_blocks` to the index manager and get rid of this method completely? We could even rename it to `get_count_vertices` while we're at it.

Reply:
Or maybe even to the info index, maybe it makes sense to be there.

Reply:
I liked it on the info-index, shouldn't we get rid of the `TransactionStorage.get_vertices_count` method then? Or is it still useful to have it?

Reply:
I tried to do this, but a test broke and I had to readd these methods in https://github.com/HathorNetwork/hathor-core/pull/496/commits/6ca5bef1670adcc95d88ad7c88ba0e1079e58ff0

I'll try to find out which test broke. Doing rebases doesn't help with this because we lose the history of tests run for each commit in the PR.

Reply:
I guess this discussions went in a different way. I was discussing about the fallback which returns an estimation instead of the actual value.